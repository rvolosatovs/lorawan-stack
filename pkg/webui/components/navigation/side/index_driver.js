// Copyright Â© 2019 The Things Network Foundation, The Things Industries B.V.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

import React from 'react'

import SideNavigation from './side'

export default function() {
  const driver = {
    component: undefined,
    location: {
      pathname: '/',
    },

    when: {
      created(props) {
        driver.component = shallow(
          <SideNavigation.WrappedComponent location={location} {...props} />,
        )

        return driver
      },
      minimized() {
        driver.get.hideButton().simulate('click')
        driver.component.update()
      },
      itemSelected(index) {
        driver.get.itemButton(index).simulate('click')
        driver.component.update()
      },
      linkSelected(index) {
        driver.get.itemLink(index).simulate('click')
        driver.component.update()
      },
      nestedLinkSelected(itemIndex, linkIndex) {
        driver.get.nestedItemLink(itemIndex, linkIndex).simulate('click')
        driver.component.update()
      },
    },
    is: {
      minimized() {
        return driver.component.state().isMinimized
      },
      expanded(index) {
        return !!driver.component.state().itemsExpanded[index].isOpen
      },
      itemExpanded(index) {
        return driver.get.item(index).props().isExpanded
      },
      itemActive(index) {
        return driver.get.item(index).props().isActive
      },
    },
    get: {
      list() {
        return driver.component.find('SideNavigationList').dive()
      },
      nestedList(index) {
        return driver.get
          .item(index)
          .dive()
          .find('CollapsableItem')
          .dive()
          .find('SideNavigationList')
      },
      hideButton() {
        return driver.component.find('[data-hook="side-nav-hide-button"]')
      },
      items() {
        return driver.get.list().children()
      },
      item(index) {
        return driver.get.items().at(index)
      },
      itemButton(index) {
        return driver.get
          .item(index)
          .dive()
          .find('CollapsableItem')
          .dive()
          .find('[data-hook="side-nav-item-button"]')
      },
      itemLink(index) {
        return driver.get
          .item(index)
          .dive()
          .find('LinkItem')
          .dive()
          .find('[data-hook="side-nav-item-link"]')
      },
      nestedItemLink(itemIndex, linkIndex) {
        return driver.get
          .nestedList(itemIndex)
          .dive()
          .find('SideNavigationItem')
          .dive()
          .find('LinkItem')
          .at(linkIndex)
          .dive()
          .find('[data-hook="side-nav-item-link"]')
      },
      itemsCount() {
        return driver.get.items().length
      },
      collapsableItemsCount() {
        return driver.get.items().findWhere(i => i.props().isCollapsable).length
      },
      expandedItemsCount() {
        return driver.get.items().findWhere(i => i.props().isExpanded).length
      },
    },
  }

  return driver
}
