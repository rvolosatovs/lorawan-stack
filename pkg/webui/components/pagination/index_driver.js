// Copyright Â© 2019 The Things Network Foundation, The Things Industries B.V.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

import React from 'react'

import Pagination from '.'

export default function() {
  const driver = {
    component: undefined,
    when: {
      created(props) {
        const wrapper = shallow(<Pagination {...props} />)

        if (wrapper.type() !== null) {
          driver.component = wrapper.dive()
          return driver
        }

        return undefined
      },
      navigatedNextPage() {
        driver.get.nextNavigationButton().simulate('click', {
          preventDefault: () => undefined,
        })
      },
      navigatedLastPage() {
        driver.get.lastPage().simulate('click', {
          preventDefault: () => undefined,
        })
      },
    },
    is: {
      pageSelected(page) {
        return driver.get.page(page).props().selected
      },
      firstPageSelected() {
        return driver.is.pageSelected(0)
      },
      lastPageSelected() {
        return driver.is.pageSelected(driver.get.pages().length - 1)
      },
      prevNavigationDisabled() {
        return driver.get.prevNavigation().hasClass('itemDisabled')
      },
      nextNavigationDisabled() {
        return driver.get.nextNavigation().hasClass('itemDisabled')
      },
    },
    get: {
      pages() {
        return driver.component.find('PageView')
      },
      page(page) {
        return driver.get.pages().at(page)
      },
      lastPage() {
        return driver.get.page(driver.get.pages().length - 1)
      },
      prevNavigation() {
        return driver.component.find('li').first()
      },
      nextNavigation() {
        return driver.component.find('li').last()
      },
      prevNavigationButton() {
        return driver.get
          .prevNavigation()
          .find('a')
          .first()
      },
      nextNavigationButton() {
        return driver.get
          .nextNavigation()
          .find('a')
          .first()
      },
    },
  }

  return driver
}
