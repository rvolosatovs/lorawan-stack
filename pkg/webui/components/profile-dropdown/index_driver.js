// Copyright Â© 2019 The Things Network Foundation, The Things Industries B.V.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

import React from 'react'

import ProfileDropdown from '.'

export default function() {
  const driver = {
    component: undefined,
    when: {
      created(props) {
        driver.component = shallow(<ProfileDropdown {...props} />)

        return driver
      },
      toggledDropdown() {
        driver.get.clickableArea().simulate('click', {
          preventDefault: () => undefined,
        })
      },
    },
    is: {
      dropdownOpen() {
        return driver.component.state('expanded')
      },
    },
    get: {
      clickableArea() {
        return driver.component.find('.container').first()
      },
      dropdown() {
        return driver.component.find('.dropdown').first()
      },
    },
  }

  return driver
}
