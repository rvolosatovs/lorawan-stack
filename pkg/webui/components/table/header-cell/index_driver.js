// Copyright Â© 2018 The Things Network Foundation, The Things Industries B.V.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

import React from 'react'

import HeaderCell from '.'

export default function () {
  const driver = {
    component: undefined,
    when: {
      created (props) {
        driver.component = shallow(
          <HeaderCell {...props} />
        )

        return driver
      },
      sortButtonPressed () {
        driver.get.sortButton().simulate('click')

        return driver
      },
    },
    get: {
      sortButton () {
        return driver.component.find('[data-hook="sort-button"]').first()
      },
    },
  }

  return driver
}
