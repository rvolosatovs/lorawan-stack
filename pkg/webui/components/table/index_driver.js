// Copyright Â© 2019 The Things Network Foundation, The Things Industries B.V.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

import React from 'react'
import Table from '.'

export default function() {
  const driver = {
    component: undefined,
    when: {
      created(props) {
        driver.component = shallow(<Table {...props} />)

        return driver
      },
      updated(props) {
        driver.component.setProps(props)
      },
      rowClicked(index) {
        driver.get.row(index + 1).simulate('click')
        driver.component.update()
      },
      sortButtonPressed(index) {
        driver.get
          .sortButton(index)
          .dive()
          .simulate('click')
        driver.component.update()
      },
    },
    is: {
      empty() {
        return driver.get.emptyMessage().exists()
      },
      paginated() {
        return driver.get.pagination().exists()
      },
      sortButtonActive(index) {
        return driver.get.sortButton(index).props().active
      },
    },
    get: {
      emptyMessage() {
        return driver.component.find('Empty').first()
      },
      headCellsCount() {
        return driver.get.headCells().length
      },
      headCells() {
        return driver.component.find('HeadCell')
      },
      dataCellsCount() {
        return driver.get.dataCells().length
      },
      dataCells() {
        return driver.component.find('DataCell')
      },
      sortButtonsCount() {
        return driver.get.sortButtons().length
      },
      sortButtons() {
        return driver.component.find('SortButton')
      },
      sortButton(index) {
        return driver.get.sortButtons().at(index)
      },
      pagination() {
        return driver.component.find('Pagination').first()
      },
      rows() {
        return driver.component.find('Row')
      },
      row(index) {
        return driver.get.rows().at(index)
      },
    },
  }

  return driver
}
