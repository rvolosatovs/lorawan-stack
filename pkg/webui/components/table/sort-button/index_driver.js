// Copyright Â© 2019 The Things Network Foundation, The Things Industries B.V.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

import React from 'react'

import SortButton from '.'

export default function() {
  const driver = {
    component: undefined,
    when: {
      created(props) {
        driver.component = shallow(<SortButton {...props} />)
      },
      buttonPressed() {
        driver.component.simulate('click')
      },
    },
    is: {
      active() {
        return driver.component.hasClass('buttonActive')
      },
      descending() {
        return driver.component.hasClass('buttonDesc')
      },
    },
  }

  return driver
}
