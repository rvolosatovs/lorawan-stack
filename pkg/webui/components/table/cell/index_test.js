// Copyright Â© 2019 The Things Network Foundation, The Things Industries B.V.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

import { getHeadCellDriver, getDataCellDriver } from './index_driver'

describe('HeadCell', function() {
  let driver = null

  beforeEach(function() {
    driver = getHeadCellDriver()
    driver.when.created({
      content: 'Head cell',
    })
  })

  it('should match snapshot', function() {
    expect(driver.component).toMatchSnapshot()
  })

  it('should be a `th` element', function() {
    expect(driver.get.cellType()).toBe('th')
  })
})

describe('DataCell', function() {
  let driver = null

  beforeEach(function() {
    driver = getDataCellDriver()
    driver.when.created({
      children: 'Data entry',
    })
  })

  it('should match snapshot', function() {
    expect(driver.component).toMatchSnapshot()
  })

  it('should be a `th` element', function() {
    expect(driver.get.cellType()).toBe('td')
  })
})
