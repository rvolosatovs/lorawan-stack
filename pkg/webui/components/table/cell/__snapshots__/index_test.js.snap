// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataCell should match snapshot 1`] = `
<Cell
  className="cellData"
  component="td"
>
  Data entry
</Cell>
`;

exports[`HeadCell should match snapshot 1`] = `
<Cell
  className="cellHead"
  component="th"
>
  <Message
    content="Head cell"
  />
</Cell>
`;
